### ISSUE TITLE
Memory leak while training with nmt_large.yml

### ISSUE THREAD

**Issue by wb14123** [google/seq2seq#193]
I'm using the config file "nmt_large.yml" in the example directory. While I'm training it with GPU, it slowly allocated all the machine memory I have (about 32G, not the GPU memory) in one day. Is this normal? How much memory will it usually use?

This is the command: 

```
python -m bin.train \
  --config_paths="
      ./example_configs/nmt_large.yml,
      ./example_configs/train_seq2seq.yml,
      ./example_configs/text_metrics_bpe.yml" \
  --model_params "
      vocab_source: $VOCAB_SOURCE
      vocab_target: $VOCAB_TARGET" \
  --input_pipeline_train "
    class: ParallelTextInputPipeline
    params:
      source_files:
        - $TRAIN_SOURCES
      target_files:
        - $TRAIN_TARGETS" \
  --input_pipeline_dev "
    class: ParallelTextInputPipeline
    params:
       source_files:
        - $DEV_SOURCES
       target_files:
        - $DEV_TARGETS" \
  --batch_size 32 \
  --train_steps $TRAIN_STEPS \
  --output_dir ./output_big
```

**Comment by mohgh** [comment]
@ wb14123 Have you solved this? training with gpu allocates all my ram.

**Comment by wb14123** [comment]
Not yet. I was going to profile the memory but didn't have the time these
days. Are you also out of machine RAM instead of GPU memory? What's your
total memory?

Mohammad Ghafourian <notifications@github.com>于2017年5月13日周六 15:08写道：

**Comment by mohgh** [comment]
Yes, my ram memory is used. My GPU only has 2 gigabytes of memory. My total RAM memory is 32 gigabytes. Even with nmt_small model params, my process will get killed after 320000 steps. With larger models, this happens faster. 
When I restart the training from checkpoint what data will be lost? Can we train the model for 1 epoch and then stop the training (RAM will be empty again) and then restart the training with saved params?

**Comment by Scitator** [comment]
Have the same problem: after 3 days of training, process already use 41G of RAM at ~54800 step.

Is there any correct way to restart training process from last checkpoint? You know, 3 days...

**Comment by mohgh** [comment]
I think this issue is related to a tensorflow bug explained here:
[https://github.com/tensorflow/tensorflow/issues/9779](url)

**Comment by wb14123** [comment]
@Scitator I think just re-run the command will use the checkpoint from the tf output directory. But I'm not sure if it will load the data from the beginning. If so, it may use the same data multi times thus makes the training over fitting.
